Ginger/GingerCoreCommon/VariablesLib/VariableNumber.cs

```diff
@@ -0,0 +1,284 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+See the License for the specific language governing permissions and 
+limitations under the License. 
+*/
+#endregion
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Amdocs.Ginger.Common;
+using Amdocs.Ginger.Common.Enums;
+using Amdocs.Ginger.Repository;
+
+namespace GingerCore.Variables
+{
+    public class VariableNumber : VariableBase
+    {
+        private string mInitialNumberValue;
+
+        private string mMinValue;
+        private string mMaxValue;
+
+        private string mPrecisionValue;
+
+        private bool? mIsDecimalValue;
+        
+        [IsSerializedForLocalRepository]
+        public bool IsDecimalValue
+        {
+            set
+            {
+                mIsDecimalValue = value;
+                OnPropertyChanged("IsDecimalValue");
+                OnPropertyChanged("InitialNumberValue");
+                OnPropertyChanged("Formula");
+            }
+            get
+            {
+                if (mIsDecimalValue == null)
+                {
+                    return false;
+                }
+                    
+                return Convert.ToBoolean(mIsDecimalValue);
+            }
+        }
+
+        [IsSerializedForLocalRepository]
+        public string MinValue
+        {
+            get 
+            {
+                if (string.IsNullOrEmpty(mMinValue))
+                {
+                    mMinValue = GetMinValue();
+                }
+                return mMinValue; 
+            }
+            set 
+            {
+                mMinValue = value;
+                OnPropertyChanged("MinValue");
+                OnPropertyChanged("InitialNumberValue");
+                OnPropertyChanged("Formula");
+            }
+        }
+
+        private string GetMinValue()
+        {
+            if (!IsDecimalValue)
+            {
+                return Int32.MinValue.ToString();
+            }
+            else
+            {
+                return float.MinValue.ToString();
+            }
+        }
+
+        [IsSerializedForLocalRepository]
+        public string MaxValue
+        {
+            get 
+            { 
+                if(string.IsNullOrEmpty(mMaxValue))
+                {
+                    mMaxValue = GetMaxValue();
+                }
+                return mMaxValue;             
+            }
+            set
+            { 
+                mMaxValue = value;
+                OnPropertyChanged("MaxValue");
+                OnPropertyChanged("InitialNumberValue");
+                OnPropertyChanged("Formula");
+            }
+        }
+
+        private string GetMaxValue()
+        {
+            if(!IsDecimalValue)
+            {
+                return Int32.MaxValue.ToString();
+            }
+            else
+            {
+                return float.MaxValue.ToString();
+            }
+        }
+
+        [IsSerializedForLocalRepository]
```

Comment:
i think you can give the default value on annotation i.e. IsSerializedForLocalRepository("2") something like this. Check for exact syntax